File: 1742/main.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	fmt.Println(countBalls(1, 10))
}

func countBalls(lowLimit int, highLimit int) int {
	ans := 0 
	m := map[int]int{}
	for i := lowLimit; i <= highLimit; i++ {
		s := strconv.Itoa(i)
		n := 0
		for _, v := range s {
			n += int(v - 48)
		}
		m[n]++
	}
	for _, v := range m {
		if v > ans {
			ans = v
		}
	}
	return ans
}

File: 1742/main.rs
use std::collections::HashMap;

struct Solution{}

impl Solution {
    pub fn count_balls(low_limit: i32, high_limit: i32) -> i32 {
        let mut c = HashMap::<i32, i32>::new();
        let mut ans = 0;
        for i in low_limit..=high_limit {
            let mut sum = 0;
            let mut x = i;
            while x > 0 {
                sum += x % 10;
                x /= 10;
            }
            if let Some(_s) = c.get(&sum) {
                c.insert(sum, _s + 1);
            } else {
                c.insert(sum, 1);
            }
            ans = ans.max(c[&sum]);
        }
        ans
    }
}

fn main() {
    println!("{:#?}", Solution::count_balls(1, 10));
}